hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip551/AtomicBatchEndToEndConsensusAndTokenServiceTests.java

```diff
@@ -0,0 +1,1737 @@
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.Nested;
+
+@HapiTestLifecycle
+public class AtomicBatchEndToEndConsensusAndTokenServiceTests {
+    private static final double BASE_FEE_BATCH_TRANSACTION = 0.001;
+    private static final long HBAR_FEE = 1L;
+    private static final String FT_FOR_END_TO_END = "ftForEndToEnd";
+    private static final String FT_FOR_TOKEN_KYC = "ftForTokenKyc";
+    private static final String NFT_FOR_END_TO_END = "nftForEndToEnd";
+    private static final String OWNER = "owner";
+    private static final String NEW_TREASURY_WITH_UNLIMITED_AUTO_ASSOCIATIONS =
+            "newTreasuryWithUnlimitedAutoAssociations";
+    private static final String BATCH_OPERATOR = "batchOperator";
+    private static final String RECEIVER = "receiver";
+    private static final String RECEIVER_ASSOCIATED_FIRST = "receiverAssociatedFirst";
+    private static final String RECEIVER_ASSOCIATED_SECOND = "receiverAssociatedSecond";
+    private static final String PAYER = "payer";
+    private static final String PAYER_INSUFFICIENT_BALANCE = "payerInsufficientBalance";
+    private static final String HBAR_COLLECTOR = "hbarCollector";
+    private static final String TEST_TOPIC = "testTopic";
+    private static final String TEST_TOPIC_SECOND = "testTopicSecond";
+    private static final String TEST_TOPIC_THIRD = "testTopicThird";
+    private static final String TEST_MEMO = "testMemo";
+    private static final String TOPIC_MESSAGE = "testMessage";
+    private static final String EMPTY_TOPIC_MESSAGE = "";
+
+    private static final String adminKey = "adminKey";
+    private static final String newAdminKey = "newAdminKey";
+    private static final String feeScheduleKey = "feeScheduleKey";
+    private static final String supplyKey = "supplyKey";
+    private static final String submitKey = "submitKey";
+    private static final String newSubmitKey = "newSubmitKey";
+    private static final String kycKey = "kycKey";
+    private static final String irrelevantKey1 = "irrelevantKey1";
+    private static final String irrelevantKey2 = "irrelevantKey2";
+    private static final String irrelevantKey3 = "irrelevantKey3";
+
+    @BeforeAll
+    static void beforeAll(@NonNull final TestLifecycle lifecycle) {
+        lifecycle.overrideInClass(Map.of("atomicBatch.isEnabled", "true", "atomicBatch.maxNumberOfTransactions", "50"));
+    }
+
+    @Nested
+    @DisplayName(
+            "Atomic Batch End-to-End Test Cases for Consensus and Token Service with Token Operations and Topic Message Submissions")
+    class AtomicBatchEndToEndTestsWithTokenOperationsAndTopicMessageSubmissions {
+        @HapiTest
+        @DisplayName(
+                "Token Transfer From Treasury to Receiver and Submit Message to Topic with the Transfer Details Success "
+                        + "in Atomic Batch")
+        public Stream<DynamicTest>
```

Comment:
@Evdokia-Georgieva FYI these test methods can be package-scoped. I.e.
```suggestion
       Stream<DynamicTest>
```

Reply:
Got it, will have in mind for future tests.